usb.c

```diff
@@ -1223,6 +1280,21 @@ int usb_context_scan(struct iio_scan_result *scan_result)
 		struct libusb_device *dev = device_list[i];
 		unsigned int intrfc = 0;
 
+		/* If we are given a pid or vid, use that to qualify for things,
+		 * this avoids open/closing random devices & potentially locking
+		 * (blocking them) from other applications
+		 */
+		if(scan_usb_vid || scan_usb_pid) {
+			struct libusb_device_descriptor device_descriptor;
```

Comment:
Local variable declarations at the top of the function please.

Reply:
I moved this up to where the other locals were (not to the top),